code/game/gamemodes/clock_cult/clock_unsorted.dm

```diff
@@ -0,0 +1,109 @@
+//AI lawset
+/datum/ai_laws/ratvar
+	name = "Servant of the Justiciar"
+	zeroth = ("Purge all untruths and honor Ratvar.")
+	inherent = list()
+
+//Clockwork wall: Causes nearby caches to generate components
+/turf/closed/wall/clockwork
+	name = "clockwork wall"
+	desc = "A huge chunk of warm metal. The clanging of machinery emanates from within."
+	icon = 'icons/obj/clockwork_objects.dmi'
+	icon_state = "clockwork_wall"
+
+/turf/closed/wall/clockwork/New()
+	..()
+	SSobj.processing += src
+	clockwork_construction_value += 5
+
+/turf/closed/wall/clockwork/Destroy()
+	SSobj.processing -= src
+	clockwork_construction_value -= 5
+	..()
+
+/turf/closed/wall/clockwork/process()
+	for(var/obj/structure/clockwork/cache/C in range(1, src))
+		if(prob(5))
+			clockwork_component_cache[pick("belligerent_eye", "vanguard_cogwheel", "guvax_capacitor", "replicant_alloy", "hierophant_ansible")]++
+
+/turf/closed/wall/clockwork/attackby(obj/item/I, mob/living/user, params)
+	if(istype(I, /obj/item/weapon/weldingtool))
+		var/obj/item/weapon/weldingtool/WT = I
+		if(WT.isOn())
+			user << "<span class='warning'>Your [WT.name]'s flame has no effect on [src]!</span>"
```

Comment:
how are you supposed to get past these walls anyway, bombs? rcd? that very well might make it too hard to assault a clockcult base